lib/features/add_property/view/widgets/add_property_type_view.dart

```diff
@@ -0,0 +1,222 @@
+                      top: HsDimens.spacing20, bottom: HsDimens.spacing4),
+                  child: Text(HatSpaceStrings.of(context).availableDate,
+                      style: Theme.of(context).textTheme.bodyMedium)),
+            ),
+            // Show DatePicker Widget;
+            ValueListenableBuilder<DateTime>(
+              valueListenable: _selectedDate,
+              builder: (context, value, child) => _DatePickerView(
+                selectedDate: value,
+                onSelectedDate: (value) {
+                  _selectedDate.value = value;
+                },
+              ),
+            ),
+          ],
+        ),
+      );
+
+  void _onPropertyTypeChanged(PropertyTypes type) => _propertyType.value = type;
+
+  @override
+  void dispose() {
+    _selectedDate.dispose();
+    _propertyType.dispose();
+
+    super.dispose();
+  }
+}
+
+class _PropertyTypeCardView extends StatelessWidget {
+  final PropertyTypes type;
+  final bool isSelected;
+  final ValueChanged<PropertyTypes> onSelected;
+
+  const _PropertyTypeCardView(
+      {required this.type,
+      required this.isSelected,
+      required this.onSelected,
+      Key? key})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) => AspectRatio(
+      aspectRatio: 1,
+      child: Card(
+          shadowColor: Colors.transparent,
+          margin: const EdgeInsets.only(top: HsDimens.spacing20),
+          elevation: 6,
+          color: isSelected ? HSColor.accent : HSColor.neutral2,
+          shape: RoundedRectangleBorder(
+              side: BorderSide(
+                  width: 1.5,
+                  color: isSelected ? HSColor.onAccent : Colors.transparent),
+              borderRadius: const BorderRadius.all(Radius.circular(8))),
+          child: InkWell(
+            borderRadius: BorderRadius.circular(8.0),
+            onTap: () => onSelected(type),
+            child: Padding(
+              padding: const EdgeInsets.symmetric(
+                  horizontal: HsDimens.spacing16, vertical: HsDimens.spacing36),
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.center,
+                children: [
+                  Expanded(
+                      child: SvgPicture.asset(
+                    type.getIconPath(),
+                  )),
+                  const SizedBox(
+                    height: 16,
+                  ),
+                  Expanded(
+                      child: Text(type.displayName,
+                          maxLines: 2,
+                          overflow: TextOverflow.ellipsis,
+                          style: Theme.of(context)
+                              .textTheme
+                              .bodyLarge
+                              ?.copyWith(fontWeight: FontWeight.w700)))
+                ],
+              ),
+            ),
+          )));
+}
+
+class _DatePickerView extends StatelessWidget {
```

Comment:
should we move this widget into share folder. I belive that we can reuse this widget

Reply:
No, this is specific to this screen's UI, and it will show `HsDatePicker` which is already shared